optimum/intel/openvino/quantization.py

```diff
@@ -744,6 +784,53 @@ def disable_progress_bar(self, disable: bool = True) -> None:
         else:
             self.model._progress_bar_config["disable"] = disable
 
+    def _prepare_feature_extraction_calibration_data(
+        self,
+        quantization_config: OVQuantizationConfigBase,
+        dataset: "Dataset",
+        seq_len: int = 128,
+    ) -> OVCalibrationDataset:
+        from optimum.intel import OVModelForFeatureExtraction, OVSentenceTransformer
+
+        self.model.compile()
+
+        num_samples = quantization_config.num_samples or 128
+        calibration_data = []
+        try:
+            self.model.request = InferRequestWrapper(
+                self.model.request,
+                calibration_data,
+                inference_result_mock={
+                    "last_hidden_state": np.empty((1,), np.float32),
+                    "token_embeddings": np.empty((1,), np.float32),
+                    "sentence_embedding": np.empty((1,), np.float32),
+                },
+            )
+
+            tokenizer = None
+            pbar = tqdm(total=num_samples, desc="Collecting calibration data")
+            for item in dataset:
+                sentence = item["text"].strip()[:seq_len]
+                if len(sentence) < seq_len:
```

Comment:
what about using padding and truncation during tokenization instead of filtering small strings by number of words. Besides that number of symbols in string is not always equal length of tokenized input (some letter may be represented by 2-4 tokens, some subwords may be represented by 1 token), this approach does not looks robust

Reply:
Thanks for the suggestion! I've added truncation based on number of tokens. I don't think padding is a good idea, because some inputs will contain too few actual non-padded tokens.